### PR TITLE
#53 Don't use craft's project.yml for migrations.

### PR DESCRIPTION
**Pull request by roelvanhintum**
Restores support for Matrix and Supertable.

### CODE REVIEW

**Comment by samhibberd** [comment]
Thanks very much for this @roelvanhintum appreciate the time you spent on it, i'm going to make the changes directly, there are a few situations not covered (there were a selection of different field / plugin handles for the craft 2 link it) and i'm also going to leave the project config logic as advised by @andris over at Craft.

**Comment by andris-sevcenko** [comment]
You definitely should not just be removing rows from the database (https://github.com/fruitstudios/craft-linkit/pull/54/commits/e207ea04aae1c8b2ffbf1888bebf44e7a021f5b0#diff-eb4ab0584b5426ababdb84f2ffadf519R38)

The purpose of looping through project configuration and removing it from there as well because updating from Craft 2 to Craft 3.1 would automatically put the incompatible plugins in the project config as well. Just removing them from the database would cause Craft to choke when trying to load the corresponding rows from the plugin table for such installations (C2 -> C3.1)

Also, updating the settings straight in the database (https://github.com/born05/craft-linkit/blob/8bc9524c7c1a26136371ef65f5063e8f940a3176/src/migrations/Install.php#L52-L55) means that Project Config won't be aware to those changes, leading to project configuration migration problems down the road because the settings in the fields table don't reflect the settings in project config.

=================

The *strongly* recommended approach is muting Project Config events and making the same changes in Project Config and database. One of the reasons for doing that over just using services that would populate the changes is to avoid conflicts in scenarios when you're running a migration AND applying project config changes at the same time.

=================

To sum up: to fix #53, the migration that updates the fields should select all the Link It fields from the database, loop through those and look at the context. If the context is "global", then update them in the `fields` project config key, if it's something else, then figure out where to update that. And, of course, update it in the DB.

**Comment by samhibberd** [comment]
Thanks for the input @andris-sevcenko actioning this now!
